webpack/assets/javascripts/react_app/components/Editor/Editor.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ToastNotification } from 'patternfly-react';
+
+import { noop } from '../../common/helpers';
+import DiffView from '../DiffView/DiffView';
+import EditorView from './components/EditorView';
+import EditorNavbar from './components/EditorNavbar';
+import EditorModal from './components/EditorModal';
+import './editor.scss';
+
+class Editor extends React.Component {
+  componentDidMount() {
+    const { initializeEditor } = this.props;
+    initializeEditor(this.props);
```

Comment:
Could you send only the relevant data? It feels risky and we might lose control over what actually sent to this method.

Reply:
can i access the state from the action itself with a selector? is it a good idea?

Reply:
You can access the state from within the action and pass it to a selector (which is one of the reason I try to push us back to selectors that recive the whole state object).

Some developers says you never actually need to do that because you can recive the state into your component props (with selectors) so you can just pass them to your actions.
I bealive in some cases it is more beneficial to do what you described and use the `getState` method.

In this case, do you think it would be more beneficial to get your data directly from the state? Isn't all the data you need already exists for you in the component props?